### PR TITLE
RDataField treats NS, CNAME as DNS string type.

### PR DESCRIPTION
**Pull request by m3047**
That's about it. At present the code only correctly handles PTR RRs, this makes it so that it also handles NS and CNAME RRs.

### CODE REVIEW

**Comment by gpotter2** [comment]
Apart from the permissions, this looks good :) Thanks

**Comment by m3047** [comment]
Re permissions, was working in a VM on OSX, and it maps the permissions. Sorry about that.

File: scapy/layers/dns.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def m2i(self, pkt, s):
         family = None
         if pkt.type == 1: # A
             family = socket.AF_INET
-        elif pkt.type == 12: # PTR
+        elif pkt.type in (2,5,12): # NS,CNAME,PTR
```
Comment (p-l-) [inline scapy/layers/dns.py]: Can you use a list here?

**Comment by p-l-** [comment]
Replaced by #1041.

**Pull request closed without merging**